File: ch4113ng3/python/main.py
def nerdify(txt):
    res = ""
    for i in txt:
        if i.lower() == "a":
            res += ("4")
        elif i.lower() == "e":
            res += ("3")
        elif i == "l":
            res += ("1")
        else:
            res += (i)
    return res

def nerdify_refactored(txt):
    return txt.translate(str.maketrans("aAeEl", "44331"))

File: ch4113ng3/javascript/main.js
function nerdify(txt){
  let res = "";
  for (let i = 0; i < txt.length; i++){
    if (txt[i] === "a" || txt[i] === "A"){
      res += "4";
    }
    else if (txt[i] === "E" || txt[i] === "e"){
      res += "3";
    }
    else if (txt[i] === "l"){ 
      res += "1";
    }
    else{
      res += txt[i];
    }
  }
  return res;
}

const nerdify = t => t.replace(/[aelAE]/g, a => ({'a': 4, "e": 3, "l": 0}[a.toLowerCase()]));
